src/main.rs

```diff
@@ -1,28 +1,22 @@
-use iroha::*;
-use std::thread;
-use std::time::Duration;
+use iroha::{config::Configuration, Iroha};
+use std::process;
 
-fn main() {
+#[async_std::main]
+async fn main() {
     println!("Hyperledgerいろは2にようこそ！");
-    // Setup data structures
-    let config = iroha::config::Configuration::from_path("config.json")
-        .expect("Failed to read configuration.");
-    // move block creation to another thread
-    let block_build_step_ms = config.block_build_step_ms;
-    thread::spawn(move || loop {
-        thread::sleep(Duration::from_millis(block_build_step_ms));
-        println!("building block");
-    });
-    // Initialize world state view (reads WSV if it already exists; don't worry, we will audit it
-    // later in the background to make sure it hasn't been tampered with)
-    // spawn auditor thread from block store to WSV
-    //TODO:
-    networking::torii::Torii::new(
-        &config.torii_url,
-        consensus::sumeragi::Sumeragi::new(model::block::Blockchain::new(
-            futures::executor::block_on(storage::kura::Kura::strict_init())
-                .expect("Failed to init Kura."),
-        )),
-    )
-    .start();
+    match Configuration::from_path("config.json") {
+        Ok(config) => match Iroha::new(config).await {
+            Ok(mut iroha) => {
+                iroha.start().await;
+            }
+            Err(e) => {
+                eprintln!("Failed to create Iroha from provided configuration: {}", e);
+                process::exit(2);
+            }
+        },
+        Err(e) => {
+            eprintln!("Failed to read configuration: {}", e);
+            process::exit(1);
+        }
```

Comment:
We can make main to return a `Result` and propogate an error with `?` here.

Reply:
Fix me if I wrong, but it will cause panic if we return Err from main?
This solution tries to avoid panics to make error logging clean.

Reply:
In [docs](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html) it's said that that `then main will exit with an error code (not 0) and print out a Debug representation of err`. Tried it in a playground, seems to work that way: [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a5244fea3e8ce5318674386473d17a64).

Reply:
No panic :)

Reply:
Great - missed that, will return result instead.

Reply:
done